server/index.js

```diff
@@ -36,6 +36,26 @@ app.get('/api/products', (req, res, next) => {
     });
 });
 
+app.get('/api/products/:productId', (req, res, next) => {
+  const id = req.params.productId;
+  const params = [id];
+  const sql = `
+    select *
+      from "products"
+      where "productId" = $1;`;
+  db.query(sql, params)
+    .then(response => {
+      if (response.rows.length === 0) {
+        next(new ClientError(`cannot ${req.method} ${req.originalUrl}`, 404));
```

Comment:
You error message can be more specific about not being able to find a matching product.